### ISSUE TITLE
Use rewriteRules to assume ".html" suffix

### ISSUE THREAD

**Issue by tohuw** [BrowserSync/browser-sync#1437]
### Issue details

I want to use rewriteRules to assume that e.g. `http://foo.bar/foo/bar` ought to be `http://foo.bar/foo/bar.html`. I'm attempting to do this with RegExp.

### Steps to reproduce/test case

    rewriteRules: [{
       match: new RegExp('\/[^/]+(?!\/)$'),
       replace: '/$1.html'
     }]

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [ 2.18.13 ]
- Node        [ 8.5.0 ]
- Npm         [ 5.4.2 ]

### Affected platforms

- [ ] linux
- [ ] windows
- [x] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other _(please specify which)_

### Browsersync use-case

- [ ] API
- [x] Gulp
- [ ] Grunt
- [ ] CLI

### for all other use-cases, (gulp, grunt etc), please show us exactly how you're using Browsersync

    const gulp = require('gulp')
    const devServer = browserSync.create()

    const gulp = require('gulp')
    const devServer = browserSync.create()

    gulp.task('watch:serve', function () {
        devServer.init({
            server: './output',
            rewriteRules: [{
            match: new RegExp('\/[^/]+(?!\/)$'),
            replace: '/$1.html'
            }]
        })
    })

**Comment by tohuw** [comment]
While not at the core of my problem is #1008 still a bug? That will obviously affect me...

**Comment by tohuw** [comment]
Alright, so I figured it out, but solving this problem with rewriteRules would be extremely non-trivial. Because it simply modifies HTML files and doesn't rewrite URLs, I need to write the rules targeted towards the hrefs in the documents. That gets tricky, because e.g. `http://foo.bar/foo` might really need to be `http://foo.bar/foo/index.html` and not `http://foo.bar/foo.html`.

What I really want is some analog to NGINX's `try_files` parameter. Closing this and opening a feature request.
